osf/models/mixins.py

```diff
@@ -2149,6 +2125,44 @@ def flag_spam(self):
             log.should_hide = True
             log.save()
 
+        if settings.SPAM_THROTTLE_AUTOBAN:
+            creator = self.creator
+            yesterday = timezone.now() - timezone.timedelta(days=1)
+            node_spam_count = creator.all_nodes.filter(
+                models.Q(spam_status=SpamStatus.FLAGGED) | models.Q(spam_status=SpamStatus.SPAM), models.Q(created__gt=yesterday)).count()
+            preprint_spam_count = creator.preprints.filter(
+                models.Q(spam_status=SpamStatus.FLAGGED) | models.Q(spam_status=SpamStatus.SPAM), models.Q(created__gt=yesterday)).count()
+
+            if (node_spam_count + preprint_spam_count) > settings.SPAM_CREATION_THROTTLE_LIMIT:
+                self.suspend_spam_user(creator)
+
+    def suspend_spam_user(self, user):
+        self.set_privacy('private', log=False, save=False)
```

Comment:
nit: this seems redundant with marking everything as spam below (we're explicitly checking to not re-mark self as private).

Also, it's probably too-much overhead for this PR, but this would be a great candidate to move to the User class instead of the SpamOverrideMixin -- it makes more sense for a User to access and udpate all of its content than for a random node/preprint to do so on behalf of a user.